DigitalLearningSolutions.Data/Services/LoginService.cs

```diff
@@ -49,7 +49,12 @@ public LoginResult AttemptLogin(string username, string password)
                 adminAccountVerificationAttemptedAndFailed &&
                 !delegateAccountVerificationSuccessful;
 
-            var adminAccountIsAlreadyLocked = unverifiedAdminUser?.IsLocked == true;
+            var userEmail = verifiedDelegateUsers.Any() ? verifiedDelegateUsers[0].EmailAddress : null;
+            var adminAccountAssociatedWithDelegateAccount =
+                userEmail == null ? null : userService.GetUsersByEmailAddress(userEmail).adminUser;

```

Comment:
`userService.GetUsersByEmailAddress` does an unnecessary query in this instance - in `userDataService.GetDelegateUsersByEmailAddress`. Is it worth adding a new method to just get the admin user?

I note that there is a separate `GetDelegateUsersByEmailAddress` method in this file, but not a `GetAdminUserByEmailAddress` one. I also note that `GetUsersByEmailAddress` checks `string.IsNullOrWhiteSpace(emailAddress)`, but `GetDelegateUsersByEmailAddress` does not.